front/src/applications/operationalStudies/hooks/useTrackSections.ts

```diff
@@ -0,0 +1,47 @@
+import { useState, useCallback } from 'react';
+
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import type { InfraObjectWithGeometry } from 'common/api/osrdEditoastApi';
+import type { AppDispatch } from 'store';
+
+export default function useCachedTrackSections(infraId: number, dispatch: AppDispatch) {
+  const [trackIds, setTrackIds] = useState<Set<string>>(new Set());
+  const [trackSections, setTrackSections] = useState<InfraObjectWithGeometry[]>([]);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  const loadTracksSectionsByIds = useCallback(
+    async (newTrackIds: string[]) => {
+      const uniqueNewIds = newTrackIds.filter((id) => !trackIds.has(id));
+      if (uniqueNewIds.length === 0) return;
+
+      const fetchedSections = await dispatch(
+        osrdEditoastApi.endpoints.postInfraByInfraIdObjectsAndObjectType.initiate({
+          infraId,
+          objectType: 'TrackSection',
+          body: uniqueNewIds,
+        })
+      ).unwrap();
+
+      setTrackIds((prev) => new Set([...prev, ...uniqueNewIds]));
+      setTrackSections((prev) => [...prev, ...fetchedSections]);
+    },
+    [trackIds, infraId, dispatch]
+  );
+
+  const getTrackSectionsByIds = useCallback(
+    async (requestedTrackIds: string[]) => {
+      setLoading(true);
+      try {
+        await loadTracksSectionsByIds(requestedTrackIds);
+      } catch (error) {
+        console.error('Failed to fetch track sections:', error);
+      } finally {
+        setLoading(false);
+      }
+      return trackSections.filter((section) => requestedTrackIds.includes(section.obj_id));
```

Comment:
> Are we sure trackSections will be already updated after the call of loadTracksSectionsByIds ?

I assumed it would, but indeed it may not be.

> Do we have to pass by a state for trackSections value ? Couldn't we just pass the content of loadTracksSectionsByIds in the try so we can access the result directly ?

We can't just pass the value without also reading the state because we only fetch new ids, but you're right this needs to be reworked a bit.